internal/dag/accessors.go

```diff
@@ -99,6 +99,9 @@ func upstreamProtocol(svc *v1.Service, port v1.ServicePort) string {
 	if protocol == "" {
 		protocol = up[strconv.Itoa(int(port.Port))]
 	}
+	if protocol == "" && port.AppProtocol != nil {
```

Comment:
Shouldn't this check be before the previous one? Since the previous one will change the value of protocol if it's empty, and this one expects protocol to be empty, so it will never get hit?